### PR TITLE
Fix TGS

### LINKED ISSUE #10450: TGS bugs no. 2
### Your GTNH Discord Username

chochom#0271

### Your Pack Version

2.1.2.4 5-dev-pre-test

### Your Server

SP

### Type of Server

Single Player

### Your Expectation

when an electric chainsaw runs out of energy it should land in the output bus.

### The Reality

At first the empty chainsaw just vanishes. Sometimes, later, it then puts out a lot of empty chainsaw get put out into the output bus at once. I am not fully sure what is needed to reproduce this.

### Your Proposal

fix it.

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [ ] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

### LINKED ISSUE #10599: Tree Growth Simulator destroy depleted chainsaw
### Your GTNH Discord Username

pulsemageadvocate

### Your Pack Version

2.1.2.4-14-pre

### Your Server

SP

### Type of Server

Single Player

### Your Expectation

Chainsaw consume 100EU every damage.
Tree Growth Simulator output chainsaw when EU < 100.

### The Reality

Chainsaw with EU >= 32(?) is not outputted, and destroyed by next damage.

### Your Proposal

fix

### Final Checklist

- [X] I have searched this issue tracker and there is nothing similar already. Posting on a closed issue saying the bug still exists will prompt us to investigate and reopen it once we confirm your report.
- [X] I can reproduce this problem consistently by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.
- [ ] I have asked other people and they confirm they also have this problem by follow the exact steps I described above, or this does not need reproducing, e.g. recipe loophole.

### PR DESCRIPTION
**Pull request by DianeXD**
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10450
close https://github.com/GTNewHorizons/GT-New-Horizons-Modpack/issues/10599

### CODE REVIEW

**Comment by Steelux8** [comment]
Please be more descriptive of what you changed here, and for what reason.

**Comment by DianeXD** [comment]
In addition to solving the above problems, now the saw will get damage every time it works
Also removed some useless code

File: src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java [hunk 1]
```diff
@@ -187,41 +172,31 @@ public boolean checkRecipe(final ItemStack aStack) {
 		this.mEfficiency = (10000 - (getIdealStatus() - getRepairStatus()) * 1000);
 		this.mEfficiencyIncrease = 10000;
 
-		// Overclock
-		if (this.mEUt <= 16) {
```
Comment (GlodBlock) [inline src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java:191]: are you sure tgs isn't allowed for overclocking recipe?

Comment (DianeXD) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java:191]: > are you sure tgs isn't allowed for overclocking recipe?

It works at a constant speed of 100 tick

Comment (GlodBlock) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java:191]: then add this info to tooltip plz

Comment (chochem) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java:191]: adding the time to the tooltip is pretty useless if one doesnt write how much it makes at that point. the quadratic formula for logs per second would be a lot more useful.

**Review by GlodBlock — APPROVED** [review]

File: src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java (hunk 1 above)
Comment (GlodBlock) [reply src/main/java/gtPlusPlus/xmod/gregtech/common/tileentities/machines/multi/production/GregtechMetaTileEntityTreeFarm.java:191]: yeah, add the formula of the output amount is also useful

**Review by draknyte1 — COMMENTED** [review]
Sapling production should not happen without fertiliser at all. Community requested this specifically, as most people don’t want to deal with saplings. It’s mostly intended for saplings used in production of Radox, which should *not* be given freely.

**Comment by DianeXD** [comment]
> Sapling production should not happen without fertiliser at all. Community requested this specifically, as most people don’t want to deal with saplings. It’s mostly intended for saplings used in production of Radox, which should _not_ be given freely.

No fertilizer will only output logs. When fertilizer is insufficient or sufficient, only output saplings.

**Pull request merged**